RecoLocalCalo/HGCalRecProducers/plugins/HGCalClusterTestProducer.cc

```diff
@@ -152,8 +164,32 @@ void HGCalClusterTestProducer::produce(edm::Event& evt,
   //   std::cout << "hydra " << names[i] << " size : " << hydra[i]->size() << std::endl;
   // }
 
-  evt.put(std::move(clusters));
-  evt.put(std::move(clusters_sharing),"sharing");
+  
+
+  auto clusterHandle = evt.put(std::move(clusters));
+  auto clusterHandleSharing = evt.put(std::move(clusters_sharing),"sharing");
+
+  edm::PtrVector<reco::BasicCluster> clusterPtrs, clusterPtrsSharing;
+  for( unsigned i = 0; i < clusterHandle->size(); ++i ) {
+    edm::Ptr<reco::BasicCluster> ptr(clusterHandle,i);
+    clusterPtrs.push_back(ptr);
+  }
+  
+  for( unsigned i = 0; i < clusterHandleSharing->size(); ++i ) {
+    edm::Ptr<reco::BasicCluster> ptr(clusterHandleSharing,i);
+    clusterPtrsSharing.push_back(ptr);
+  }
+
+  std::unique_ptr<std::vector<reco::HGCalMultiCluster> > 
+    multiclusters( new std::vector<reco::HGCalMultiCluster> ), 
+    multiclusters_sharing( new std::vector<reco::HGCalMultiCluster> );
+
+  *multiclusters = std::move(multicluster_algo->makePreClusters(clusterPtrs));
```

Comment:
remind me please what is the meaning of "Pre" here ?
Is there going to be another clustering algorithm on top of this?